src/azul/terra.py

```diff
@@ -195,11 +199,6 @@ def _oauthed_request(self,
                       trunc_ellipses(response.data, 256))
         return response
 
-    def get_access_token(self) -> str:
-        credentials = self.credentials.with_scopes(self.oauth_scopes)
```

Comment:
Maybe I'm missing something here, but it looks like the old way didn't use the bigquery.readonly scope. If I recall correctly from the call, you said it used a more general scope, but it looks like that's not the case?

Reply:
When the `Credentials` object is instantiated, it has no scopes, and is implicitly authorized for everything the Service Account has access to. The call to `with_scopes` reduces that authorization to the provided scopes. Note that `with_scopes` returns a new `Credentials` and does not mutate the original.

Prior to these changes, `with_scopes` was only called on the line you've selected, and only the local variable in `SAMClient.get_access_token` ever had its scopes limited. Thus, when `TDRClient` instantiated its `BigQuery` client, [here](https://github.com/DataBiosphere/azul/blob/d66e82be0c35510a4a865ae4599cfd85ec02e7d1/src/azul/terra.py#L314), it is providing it with the original, unscoped credentials, which implicitly contains the `bigquery.readonly` scope (because it has _no_ explicit scopes).

My changes move the call to `with_scopes` into the `credentials` property so that all `TerraClient` instances only ever have the credential scopes that they declare at all call sites. As a consequence, we must now declare certain scopes that previously were granted implicitly.

Reply:
I see. Thanks for explaining this so thoroughly!